addons/account/models/account_account.py

```diff
@@ -762,12 +768,21 @@ def _toggle_reconcile_to_false(self):
         if partial_lines_count > 0:
             raise UserError(_('You cannot switch an account to prevent the reconciliation '
                               'if some partial reconciliations are still pending.'))
-        query = """
-            UPDATE account_move_line
-                SET amount_residual = 0, amount_residual_currency = 0
-            WHERE full_reconcile_id IS NULL AND account_id IN %s
-        """
-        self.env.cr.execute(query, [tuple(self.ids)])
+        with Lines.modifying_model(
+            inputs=["full_reconcile_id"],
+            outputs=["amount_residual", "amount_residual_currency"],
+        ) as m:
+            query = """
+                UPDATE account_move_line
+                    SET amount_residual = 0, amount_residual_currency = 0
+                WHERE full_reconcile_id IS NULL AND account_id IN %s
+                RETURNING id
+            """
+            self.env.cr.execute(query, [tuple(self.ids)])
+            if self.env.cr.rowcount:
+                m.records = Lines.browse(i for [i] in self.env.cr.fetchall())
+            else:
+                m.records = Lines.browse(())
```

Comment:
It is technically correct, but in practice no one cares...
Just invalidate the whole model.

Reply:
You are potentially going to call modified several hundreds of thousands of records, and we don't care about it in practice

It never caused an issue, and now you are adding a potential performance issue

Reply:
It is not illegal, but we don't change the `reconcile` flag of payable/receivable accounts (it's always `True`), and it only makes sense to have dependencies related to those accounts.

Absolute correctness is not always better